app/src/main/java/com/xpay/kotlinutils/XpayUtils.kt

```diff
@@ -91,19 +93,21 @@ object XpayUtils {
         // check for API settings
         checkAPISettings()
 
-        variableAmountID?.let { body= RequestBody(it) }
-        communityId?.let { body?.community_id = it }
+        val bodyPay = PayRequestBody()
+
+        variableAmountID?.let { bodyPay.variable_amount_id = it }
+        communityId?.let { bodyPay.community_id = it }
 
         // Payment method
         payUsing?.let {
             if (it in activePaymentMethods) {
-                body?.pay_using = it
+                bodyPay.pay_using = it
```

Comment:
pay_using here must be string 
 requestBody["pay_using"] = it.toString().toLowerCase(Locale.ROOT)